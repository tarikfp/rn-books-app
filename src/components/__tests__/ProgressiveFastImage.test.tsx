import { render } from '@testing-library/react-native';
import React from 'react';
import ProgressiveImage from '~components/ProgressiveFastImage';

describe('ProgressiveImage Component', () => {
  it('matches snapshot', () => {
    const mockSource = { uri: 'mock-image-uri' };
    const { toJSON } = render(
      <ProgressiveImage
        source={mockSource}
        blurRadius={10}
        thumbnailSource={mockSource}
        resizeMode="cover"
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
