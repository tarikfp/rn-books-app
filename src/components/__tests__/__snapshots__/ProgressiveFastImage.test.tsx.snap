// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressiveImage Component matches snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "transparent",
      },
    ]
  }
>
  <Image
    blurRadius={10}
    collapsable={false}
    onLoad={[Function]}
    source={
      {
        "uri": "mock-image-uri",
      }
    }
    style={
      {
        "opacity": 0,
      }
    }
  />
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
  >
    <FastImageView
      collapsable={false}
      defaultSource={null}
      onFastImageError={[Function]}
      onFastImageLoad={[Function]}
      onFastImageLoadEnd={[Function]}
      resizeMode="cover"
      source={null}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
